packages/langium/src/lsp/workspace-symbol-provider.ts

```diff
@@ -0,0 +1,70 @@
+/******************************************************************************
+ * Copyright 2023 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import type { DocumentSymbol, WorkspaceSymbol, WorkspaceSymbolParams } from 'vscode-languageserver';
+import type { LangiumSharedServices } from '../services';
+import type { ServiceRegistry } from '../service-registry';
+import type { LangiumDocuments } from '../workspace';
+import { CancellationToken, TextDocumentIdentifier } from 'vscode-languageserver';
+import { interruptAndCheck, type MaybePromise } from '../utils/promise-util';
+
+export interface WorkspaceSymbolProvider {
+    getSymbols(params: WorkspaceSymbolParams, cancelToken?: CancellationToken): MaybePromise<WorkspaceSymbol[]>;
+    resolveSymbol?(symbol: WorkspaceSymbol, cancelToken?: CancellationToken): MaybePromise<WorkspaceSymbol>;
+}
+
+export class DefaultWorkspaceSymbolProvider implements WorkspaceSymbolProvider {
+
+    protected readonly serviceRegistry: ServiceRegistry;
+    protected readonly documents: LangiumDocuments;
+
+    constructor(services: LangiumSharedServices) {
+        this.serviceRegistry = services.ServiceRegistry;
+        this.documents = services.workspace.LangiumDocuments;
+    }
+
+    async getSymbols(params: WorkspaceSymbolParams, cancelToken = CancellationToken.None): Promise<WorkspaceSymbol[]> {
```

Comment:
The [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_symbol) says:

> The workspace symbol request is sent from the client to the server to list project-wide symbols

Wouldn't it be more accurate to use `IndexManager.allElements` to find the workspace symbols?

Reply:
Yeah, that probably makes more sense 👍